pkg/network/sriov/sriov.go

```diff
@@ -19,4 +19,84 @@
 
 package sriov
 
+import (
+	"fmt"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	k8sv1 "k8s.io/api/core/v1"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/multus"
+	"kubevirt.io/kubevirt/pkg/network/podinterfacenamescheme"
+)
+
 const AliasPrefix = "sriov-"
+
+func PrepareSriovNetworkPciMap(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
+	networkPciMap, err := mapNetworkNameToPciAddress(vmi, pod)
+	if err != nil {
+		return err
+	}
+
+	if networkPciMap == nil {
+		return fmt.Errorf("network-pci map empty")
+	}
+	networkPci := multus.NewNetworkPciMap(networkPciMap)
+	err = networkPci.SetAnnotationInPod(pod)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func FilterVMISRIOVInterfaces(vmi *virtv1.VirtualMachineInstance) []virtv1.Interface {
+	var interfaces []virtv1.Interface
+	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+		if iface.SRIOV != nil {
+			interfaces = append(interfaces, iface)
+		}
+	}
+	return interfaces
+}
+
+func mapNetworkNameToPciAddress(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) (map[string]string, error) {
+	multusInterfaceNameToNetworkStatusMap, err := mapMultusInterfaceNameToNetworkStatus(pod)
+	if err != nil {
+		return nil, err
+	}
+	podInterfaceNamer := podinterfacenamescheme.NewPodInterfaceNamer(vmi)
+
+	networkPciMap := map[string]string{}
+	for _, sriovIface := range FilterVMISRIOVInterfaces(vmi) {
+		networkName := sriovIface.Name
+		multusInterfaceName := podInterfaceNamer[networkName]
+		networkStatusEntry := multusInterfaceNameToNetworkStatusMap[multusInterfaceName]
+		if networkStatusEntry.DeviceInfo == nil || networkStatusEntry.DeviceInfo.Pci == nil {
+			return nil, fmt.Errorf("SRIOV interface %s has no corresponding device-info/pci-address "+
+				"in the appropriate network status annotations", networkName)
+		}
+
+		pciAddress := networkStatusEntry.DeviceInfo.Pci.PciAddress
+		if pciAddress == "" {
+			return nil, fmt.Errorf("failed to associate pci-address to SRIOV interface %s", networkName)
+		}
+		networkPciMap[networkName] = pciAddress
+	}
+	return networkPciMap, nil
```

Comment:
That's a good question.
Currently network-status annotation does not give partial data, but it may in the future (? maybe bug or some new feature).
But I think it may lead us to dark corners.. If there is no 1:1 mapping then what will the dom.xml look like? will it combust? We need to examine this and possible provide a hybrid solution, perhaps in a future PR..

Reply:
>But I think it may lead us to dark corners.. If there is no 1:1 mapping then what will the dom.xml look like? will it combust?

The domain XML should be generated correctly, there will be 5 HostDevices objects, 3 of them with the correct PCI 
address, the other two might get the wrong PCI address (50% chance).
There should be no problem with sterilizing the HostDevices object to XML.